autogen/agentchat/contrib/gpt_assistant_agent.py

```diff
@@ -57,19 +59,15 @@ def __init__(
         """
 
         self._verbose = kwargs.pop("verbose", False)
+        openai_client_cfg, openai_assistant_cfg = self.process_assistant_config(llm_config, assistant_config)
+
         super().__init__(
-            name=name, system_message=instructions, human_input_mode="NEVER", llm_config=llm_config, **kwargs
+            name=name, system_message=instructions, human_input_mode="NEVER", llm_config=openai_client_cfg, **kwargs
         )
 
-        if llm_config is False:
-            raise ValueError("llm_config=False is not supported for GPTAssistantAgent.")
-        # Use AutooGen OpenAIWrapper to create a client
-        openai_client_cfg = copy.deepcopy(llm_config)
-        # Use the class variable
-        model_name = GPTAssistantAgent.DEFAULT_MODEL_NAME
-
         # GPTAssistantAgent's azure_deployment param may cause NotFoundError (404) in client.beta.assistants.list()
         # See: https://github.com/microsoft/autogen/pull/1721
+        model_name = GPTAssistantAgent.DEFAULT_MODEL_NAME
```

Comment:
```suggestion
        model_name = self.DEFAULT_MODEL_NAME
```
To allow for subclassing and overwriting the constant.